mantidproject/mantid
# Indirect - Iqt Monte Carlo Progress bar and option to skip error calculation

**Description of work.**
This PR adds an checkbox in the Iqt Tab of `Indirect Data Analysis` which allows the user to skip the calculation of Monte Carlo errors.
This PR also adds a progress bar which lets the user know how far through the error calculation it is.

**To test:**
1. Go to `Interfaces`->`Indirect Data Analysis`->`I(Q, t)` Tab
2. Load in the data below
3. Uncheck `Calculate Errors` - ensure the `Number of Iterations` spinbox is disabled
4. Click `Run`
5. The calculation should be quick, and the output workspace should have zero for all their errors

- Repeat again but this time have `Calculate Errors` checked. Click `Run` and make sure a progress bar in the bottom right of the screen indicating that errors are being calculated appears at some point during the calculation. This calculation will take longer. Then check that errors are displayed in the output workspace. 

**Test Files**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2374049/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2374050/irs26173_graphite002_res.zip)


Fixes #23474

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
